VideoLocker/src/main/java/org/edx/mobile/loader/CoursesAsyncLoader.java

```diff
@@ -52,6 +45,18 @@ public CoursesAsyncLoader(Context context) {
         return result;
     }
 
+    private List<EnrolledCoursesResponse> getEnrolledCourses() {
+        List<EnrolledCoursesResponse> ret = null;
+
+        try {
+            ret = environment.getServiceManager().getEnrolledCourses(false);
+        } catch (Exception e) {
```

Comment:
Ah, that’s where the offending code is. Thanks!


On Mon, May 23, 2016 at 11:13 AM, Christopher Lee <notifications@github.com>
wrote:

> In VideoLocker/src/main/java/org/edx/mobile/loader/CoursesAsyncLoader.java
> <https://github.com/edx/edx-app-android/pull/707#discussion_r64238701>:
>
> > @@ -52,6 +45,18 @@ public CoursesAsyncLoader(Context context) {
> >          return result;
> >      }
> >
> > +    private List<EnrolledCoursesResponse> getEnrolledCourses() {
> > +        List<EnrolledCoursesResponse> ret = null;
> > +
> > +        try {
> > +            ret = environment.getServiceManager().getEnrolledCourses(false);
> > +        } catch (Exception e) {
>
> This generic try-catch is improperly handling an 401 oauth exception.
>
> We want to use
> https://github.com/edx/edx-app-android/blob/master/VideoLocker/src/main/java/org/edx/mobile/user/UserAPI.java#L71
>
> instead of the old non retrofit api.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-app-android/pull/707/files/60acabaaa6e94c39c14034bc24dac065b42cc39b#r64238701>
>



-- 
Regards,

Marc DiNino
Android Engineer
EDX